main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -97,4 +98,37 @@ object TypeReduction {
     case Type.Alias(cst, args, t, _) => simplify(t, renv0, loc)
   }
 
+  def simplifyJava(tpe: Type, loc: SourceLocation)(implicit flix: Flix): Result[(Type, Boolean), TypeError] =
+    tpe.typeConstructor match {
+      case Some(TypeConstructor.MethodReturnType(m, n)) =>
+        val targs = tpe.typeArguments
+        resolveMethodReturnType(targs.head, m, targs.tail, loc) match {
+          case ResolutionResult.Resolve(t) => Result.Ok((t, true))
+          case ResolutionResult.NotFound() => ???
+          case ResolutionResult.NoProgress() => Result.Ok((tpe, false))
+        }
+      case _ => Result.Ok((tpe, false))
+    }
+
+  def resolveMethodReturnType(thisObj: Type, method: String, ts: List[Type], loc: SourceLocation)(implicit flix: Flix): ResolutionResult =
+    thisObj match {
+      case Type.Cst(TypeConstructor.Str, loc2) =>
+        val clazz = classOf[String]
+        // Filter out candidate methods by method name
+        // NB: this considers also static methods
+        val candidateMethods = clazz.getMethods.filter(m => m.getName == method)
+          .filter(m => m.getParameterCount == 0) // Type.Cst case, no parameter???
+
+        ResolutionResult.Resolve(Type.getFlixType(candidateMethods.head.getReturnType)) // For now, arbitrarily return the first candidate method
+      case _ => ResolutionResult.NoProgress()
+    }
+
+
+  sealed trait ResolutionResult
+  object ResolutionResult {
+    case class Resolve(tpe: Type) extends ResolutionResult
```

Comment:
Resolve -> Resolved
NotFound -> MethodNotFound

NotProgress can be a case object. (Ask Jonathan)